File: 20. Valid Parentheses/ValidParentheses.cpp
class Solution {
public:
    bool isValid(string s) {
        stack<char> chars;
        for (char ch : s){
            if(ch == '(' || ch == '{' || ch == '['){
                chars.push(ch);
            }else{
                if(chars.empty() ||
                    ch == ')' && chars.top() != '(' ||
                    ch == ']' && chars.top() != '[' ||
                    ch == '}' && chars.top() != '{'
                ){
                    return false;
                }
                chars.pop();
            }
        }
        return chars.empty();
    }
};

File: 20. Valid Parentheses/ValidParentheses.cs
public class Solution {
    public bool IsValid(string s) {
    
    Stack<char> stack = new Stack<char>();

        foreach(char c in s){
            if (c == '(' || c == '[' || c == '{') {
            stack.Push(c);
            } else {
                if (!stack.Any()) {
                    return false;
                }

                if ((c == ')' && stack.Peek() == '(') ||
                    (c == ']' && stack.Peek() == '[') ||
                    (c == '}' && stack.Peek() == '{')) {
                    stack.Pop();
                }else{
                    return false;
                }
            }
        }
        return !stack.Any();
    }
}